### PR TITLE
Add support for array and null property values, add support for disabled api key

### PR DESCRIPTION
**Pull request by ManuDeBuck**

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -86,8 +86,17 @@ private function check(ResponseInterface $response)
...
+            );
+        }
+
+        if ($response->getStatusCode() === 403) {
+            return new CallResult(
+                false,
+                false,
+                $this->getRemainingRequests($response),
+                $this->getMaxRequests($response),
+                ["the api key is disabled"]
```
Comment (hansott) [inline src/Client.php]: Can we use the error message from the API?

File: src/Client.php [hunk 2]
```diff
@@ -107,8 +116,7 @@ private function check(ResponseInterface $response)
                 true,
                 $this->getRemainingRequests($response),
                 $this->getMaxRequests($response),
-                ["rate limited"],
-                null
+                ["rate limited"]
```
Comment (hansott) [inline src/Client.php]: See other comment 😬

File: src/Client.php [hunk 3]
```diff
@@ -485,8 +496,8 @@ public function upsertAccount(array $account): CallResult
         $payload = [
             "identification" => $this->accountIdentifiersToArray(
                 new AccountIdentified(
-                    isset($account["accountId"]) ? (string) $account["accountId"] :  null,
-                    isset($account["domain"]) ? (string) $account["domain"] : null
+                    isset($account["accountId"]) ? (string)$account["accountId"] : null,
+                    isset($account["domain"]) ? (string)$account["domain"] : null
```
Comment (hansott) [inline src/Client.php]: Did you manage to fix the styling with `vendor/bin/phpcbf --standard=PSR12 src`?

**Review by hansott — COMMENTED** [review]
🚀

File: src/Client.php (hunk 3 above)
Comment (ManuDeBuck) [reply src/Client.php]: image

:+1:

File: src/Client.php [hunk 4]
```diff
@@ -65,29 +65,38 @@ private function getMaxRequests(ResponseInterface $response): int
     {
         $values = $response->getHeader("x-ratelimit-limit");
 
-        return count($values) > 0 ? (int) $values[0] : 0;
+        return count($values) > 0 ? (int)$values[0] : 0;
```
Comment (hansott) [inline src/Client.php]: Can you leave the space behind the casting? Helps for readability 😇

**Pull request merged**